bundles/org.openhab.binding.androiddebugbridge/src/main/java/org/openhab/binding/androiddebugbridge/internal/AndroidDebugBridgeDevice.java

```diff
@@ -0,0 +1,329 @@
+        // Try to get players info from audio dump
+        String audioDump = runAdbShell("dumpsys", "audio", "|", "grep", "ID:", "&&", "sleep", "0.3");
+        if (audioDump.length() == 0) {
+            // Fallback to media session dump
+            String devicesResp = runAdbShell("dumpsys", "media_session", "|", "grep", "PlaybackState", "&&", "sleep",
+                    "0.3");
+            return devicesResp.contains("PlaybackState {state=3");
+        }
+        return audioDump.contains("state:started");
+    }
+
+    public VolumeInfo getMediaVolume() throws IOException, AndroidDebugBridgeDeviceNotConnectedException,
+            InterruptedException, AndroidDebugBridgeDeviceReadException {
+        return getVolume(ANDROID_MEDIA_STREAM);
+    }
+
+    public void setMediaVolume(int volume)
+            throws IOException, AndroidDebugBridgeDeviceNotConnectedException, InterruptedException {
+        setVolume(ANDROID_MEDIA_STREAM, volume);
+    }
+
+    private void setVolume(int stream, int volume)
+            throws IOException, AndroidDebugBridgeDeviceNotConnectedException, InterruptedException {
+        runAdbShell("media", "volume", "--show", "--stream", String.valueOf(stream), "--set", String.valueOf(volume));
+    }
+
+    public String getModel() throws IOException, AndroidDebugBridgeDeviceNotConnectedException, InterruptedException,
+            AndroidDebugBridgeDeviceReadException {
+        return getDeviceProp("ro.product.model");
+    }
+
+    public String getAndroidVersion() throws IOException, AndroidDebugBridgeDeviceNotConnectedException,
+            InterruptedException, AndroidDebugBridgeDeviceReadException {
+        return getDeviceProp("ro.build.version.release");
+    }
+
+    public String getBrand() throws IOException, AndroidDebugBridgeDeviceNotConnectedException, InterruptedException,
+            AndroidDebugBridgeDeviceReadException {
+        return getDeviceProp("ro.product.brand");
+    }
+
+    public String getSerialNo() throws IOException, AndroidDebugBridgeDeviceNotConnectedException, InterruptedException,
+            AndroidDebugBridgeDeviceReadException {
+        return getDeviceProp("ro.serialno");
+    }
+
+    private String getDeviceProp(String name) throws IOException, AndroidDebugBridgeDeviceNotConnectedException,
+            InterruptedException, AndroidDebugBridgeDeviceReadException {
+        var propValue = runAdbShell("getprop", name, "&&", "sleep", "0.3").replace("\n", "").replace("\r", "");
+        if (propValue.length() == 0) {
+            throw new AndroidDebugBridgeDeviceReadException("Unable to get device property");
+        }
+        return propValue;
+    }
+
+    private VolumeInfo getVolume(int stream) throws IOException, AndroidDebugBridgeDeviceNotConnectedException,
+            InterruptedException, AndroidDebugBridgeDeviceReadException {
+        String volumeResp = runAdbShell("media", "volume", "--show", "--stream", String.valueOf(stream), "--get", "|",
```

Comment:
```suggestion
        String volumeResp = runAdbShell("media", "volume", "--show", "--stream", stream, "--get", "|",
```

Reply:
Seems like I can't apply this change. runAdbShell expect this "String... args".

Reply:
You're right, that won't work.